### PR TITLE
stdlib: Reduce test file parsing overhead

### PR DESCRIPTION
**Pull request by Yethal**
# Description

Retrieving tests and their annotation no longer uses nu --ide-ast spawned in a subprocess which should reduce test runner startup time.
# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

File: crates/nu-std/std/testing.nu [hunk 1]
```diff
@@ -75,7 +55,7 @@ def get-annotated [
 # Annotations that allow multiple functions are of type list<string>
 # Other annotations are of type string
 # Result gets merged with the template record so that the output shape remains consistent regardless of the table content
-def create-test-record [] {
+def create-test-record [] nothing -> record<before-each: string, after-each: string, before-all: string, after-all: string, test: list<string>, test-skip: list<string>> {
```
Comment (amtoine) [inline crates/nu-std/std/testing.nu:58]: we really need to support writing things like
```nushell
def create-test-record [] nothing -> record<
    before-each: string,
    after-each: string,
    before-all: string,
    after-all: string,
    test: list<string>,
    test-skip: list<string>
> {
```
:eyes:

**Review by amtoine — APPROVED** [review]
i could test this, try with an error in the tests of the standard library and everything looks to be working as before :ok_hand: 

might seem a little faster, hard to tell by eyeballing, but it makes the code smaller and easier to read i figures, nice job @Yethal :pray:

**Comment by amtoine** [comment]
and you've added type annotations, which i love :star_struck:

File: crates/nu-std/std/testing.nu (hunk 1 above)
Comment (Yethal) [reply crates/nu-std/std/testing.nu:58]: Either that or custom types

Comment (amtoine) [reply crates/nu-std/std/testing.nu:58]: would love to see this :relieved:

**Pull request merged**